NuRadioReco/modules/io/LOFAR/rawTBBio.py

```diff
@@ -0,0 +1,972 @@
+   There is a class below that can combine multiple files (even from different stations)
+#. It is entirely possible that one file could contain multiple stations
+   This feature is not used, so I assume that it isn't a problem (for now)
+#. Each Station has unknown clock offsets. Technically the core stations are all on one clock, but there are some
+   unknown cable delays. This is a difficult  problem, not handled here
+#. Each Antenna doesn't necessarily start reading data at precisely the same time.
+   The code below picks the latest start time so this problem can be ignored by the end user
+#. The software that inserts metadata (namely antenna positions and calibrations) sometimes "forgets" to do its job
+   The code below will automatically read the metadata from other files when necessary
+#. LOFAR is constantly changing
+   So..keeping code up-to-date and still backwards compatible will be an interesting challenge
+#. LOFAR only has 96 RCUs (receiver control units) per station (at the moment).
+   Each RCU is essentially one digitizer. Each antenna needs two RCS to record both polarizations. The result is only
+   1/3 of the antennas can be read out each time.
+
+   LOFAR keeps track of things with two ways. First, the data is all referred to by its RCUid. 0 is the 0th RCU, ect...
+   However, which antenna corresponds to which RCU depends on the antennaSet. For LOFAR-LIM the antenna set will
+   generally be "LBA_OUTER". This could change, and sometimes the antenna set is spelled wrong in the data files. (This
+   should be handled here though)
+
+   In the code below each RCU is referred to by ANTENNA_NAME or antennaID. These are the same thing (I think). They are
+   however, a misnomer, as they actually refer to the RCU, not antenna. The specific antenna depends on the antenna
+   set. For the same antenna set, however, the ANTENNA_NAME will always refer to the same antenna.
+
+   Each ANTENNA_NAME is a string of 9 digits. First three is the station ID (not name!), next three is the group (no
+   idea, don't ask), final 3 is the RCU id
+
+   For LBA_INNER data set, even RCU ids refer to X-polarized dipoles and odd RCU ids refer to Y-polarized dipoles. This
+   is flipped for LBA_OUTER antenna set. X-polarization is NE-SW, and Y-polarization is NW-SE. antenna_response.py,
+   which handles the antenna function, assumes the data is LBA_OUTER.
+
+"""
+
+import datetime
+import os
+import logging
+
+import h5py
+import numpy as np
+
+from NuRadioReco.modules.base import module
+
+import NuRadioReco.modules.io.LOFAR.rawTBBio_metadata as md
+import NuRadioReco.modules.io.LOFAR.rawTBBio_utilities as util
+
+
+logger = module.setup_logger(level=logging.WARNING)
+
+
+# nyquist_zone = {'LBA_10_90' : 1, 'LBA_30_90' : 1, 'HBA_110_190' : 2, 'HBA_170_230' : 3, 'HBA_210_250' : 3}
+conversion_dict = {
+    "": 1.0,
+    "kHz": 1000.0,
+    "MHz": 10.0 ** 6,
+    "GHz": 10.0 ** 9,
+    "THz": 10.0 ** 12,
+}
```

Comment:
Yeah, so this module is actually an import of a standalone library that is maintained by Brian Hare and is heavily tied to way we receive our data from ASTRON. I could change this (and many other things), but as soon as we would want to update this with a newer version of Brian's code, all of it would have to be changed again. So I prefer to keep this as is, and convert everything to the NRR unit system during read-in (where we use the functions from this module to read our data into the Event).